presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java

```diff
@@ -0,0 +1,136 @@
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkSTArea
+{
+    @Benchmark
+    public Object stSphericalArea(BenchmarkData data)
+    {
+        return GeoFunctions.stSphericalArea(data.geography);
+    }
+
+    @Benchmark
+    public Object stSphericalArea500k(BenchmarkData data)
+    {
+        return GeoFunctions.stSphericalArea(data.geography500k);
+    }
+
+    @Benchmark
+    public Object stArea(BenchmarkData data)
+    {
+        return GeoFunctions.stArea(data.geometry);
+    }
+
+    @Benchmark
+    public Object stArea500k(BenchmarkData data)
+    {
+        return GeoFunctions.stArea(data.geometry500k);
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private Slice geometry;
+        private Slice geometry500k;
+        private Slice geography;
+        private Slice geography500k;
+
+        @Setup
+        public void setup()
+                throws IOException
+        {
+            geometry = stGeometryFromText(utf8Slice(loadPolygon("large_polygon.txt")));
+            geometry500k = stGeometryFromText(utf8Slice(createPolygon(500000)));
+            geography = toSphericalGeography(geometry);
+            geography500k = toSphericalGeography(geometry500k);
+        }
+    }
+
+    public static void main(String[] args)
+            throws IOException, RunnerException
+    {
+        // assure the benchmarks are valid before running
+        verify();
+
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(".*" + BenchmarkSTArea.class.getSimpleName() + ".*")
+                .build();
+        new Runner(options).run();
+    }
+
+    @Test
+    public static void verify() throws IOException
+    {
+        BenchmarkData data = new BenchmarkData();
+        data.setup();
+        BenchmarkSTArea benchmark = new BenchmarkSTArea();
+
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea(data) / 3.659E8), 1000);
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea500k(data) / 38842273735.0), 1000);
+        assertEquals(benchmark.stArea(data), 0.05033099592771004);
+        assertEquals(Math.round(1000 * (Double) benchmark.stArea500k(data) / Math.PI), 1000);
+    }
+
+    private static String createPolygon(double numVertices)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("POLYGON((");
+        String separator = "";
+        for (int i = 0; i < numVertices; i++) {
+            double angle = i * Math.PI * 2 / numVertices;
+            sb.append(separator);
+            sb.append(Math.cos(angle));
+            sb.append(" ");
+            sb.append(Math.sin(angle));
+            separator = ",";
```

Comment:
Instead of setting separator to `","` after the first iteration, it'd be clearer to append it conditionally if `i > 0`. 

Alternatively, this could be written as:

```java
IntStream.range(0, numVertices)
        .mapToObj(i -> {
            double angle = i * Math.PI * 2 / numVertices;
            return Math.cos(angle) + " " + Math.sin(angle);
        })
        .collect(Collectors.joining(","));
```